Architecture/TodoApp/android/core/src/main/java/com/novoda/todoapp/tasks/data/model/Tasks.java

```diff
@@ -53,9 +53,13 @@ public boolean isEmpty() {
         return internalMap().isEmpty();
     }
 
-    public boolean isMostRecentAction(SyncedData<Task> syncedData) {
-        SyncedData<Task> currentSyncData = internalMap().get(syncedData.data().id());
-        return currentSyncData == null || syncedData.lastSyncAction() >= currentSyncData.lastSyncAction();
+    public boolean hasNoTaskMoreRecentThan(SyncedData<Task> syncedData) {
+        return hasNoTaskMoreRecentThan(syncedData.data().id(), syncedData.lastSyncAction());
+    }
+
+    public boolean hasNoTaskMoreRecentThan(Id id, long lastSyncAction) {
```

Comment:
Previously it was hard to tell if it returned true if the `Tasks` contained the more recent task, or if the `SyncedData` passed in was more recent. `Tasks.isMostRecentAction(syncedData)` ... which is more recent, the `Tasks` or the `syncedData` ? It's confusing.

Perhaps `noActionMoreRecenThan` would be better.